pkg/controller.v2/controller_pod.go

```diff
@@ -276,7 +277,35 @@ func (tc *TFJobController) addPod(obj interface{}) {
 // If the labels of the pod have changed we need to awaken both the old
 // and new replica set. old and cur must be *v1.Pod types.
 func (tc *TFJobController) updatePod(old, cur interface{}) {
-	// TODO(CPH): handle this gracefully.
+	curPod := cur.(*v1.Pod)
+	oldPod := old.(*v1.Pod)
+	if curPod.ResourceVersion == oldPod.ResourceVersion {
+		// Periodic resync will send update events for all known pods.
+		// Two different versions of the same pod will always have different RVs.
+		return
+	}
+
+	curControllerRef := metav1.GetControllerOf(curPod)
+	oldControllerRef := metav1.GetControllerOf(oldPod)
+	controllerRefChanged := !reflect.DeepEqual(curControllerRef, oldControllerRef)
+	if controllerRefChanged && oldControllerRef != nil {
```

Comment:
Is this used for updating `v1alpha1` to `v1alpha2`?
IOW, why does `ControllerRef` changed?

Reply:
We could remove the check here. It should not happen in our case. I just copied it from job controller. WDYT

Reply:
The pods maybe modified by other controllers by some mistakes.
So I think keep this check is more better :)